lxd/device/disk.go

```diff
@@ -2540,55 +2540,32 @@ func (d *disk) generateVMConfigDrive() (string, error) {
 
 	instanceConfig := d.inst.ExpandedConfig()
 
-	// Use an empty vendor-data file if no custom vendor-data supplied.
-	vendorDataKey := "cloud-init.vendor-data"
-	vendorData, ok := instanceConfig[vendorDataKey]
-	if !ok {
-		vendorDataKey = "user.vendor-data"
-		vendorData = instanceConfig[vendorDataKey]
-		if vendorData == "" {
-			vendorData = "#cloud-config\n{}"
-		}
+	// Get raw data from instance config.
+	vendorDataKey := cloudinit.GetEffectiveConfigKey(instanceConfig, "vendor-data")
+	userDataKey := cloudinit.GetEffectiveConfigKey(instanceConfig, "user-data")
+	vendorData, userData := cloudinit.GetResultingCloudConfig(instanceConfig, vendorDataKey, userDataKey, d.inst.Name(), d.inst.Project().Name)
```

Comment:
Could we have the logic for `GetEffectiveConfigKey` moved inside `GetResultingCloudConfig` and then avoid the caller needing to figure this stuff out by just having a `requestedKey` argument that the the `cloudinit` package figures out what needs to be returned?

Reply:
Rather than "Resulting" lets use "Effective" here as that aligns more with other concepts in LXD (effective project for example).